### ISSUE TITLE
from android.support to androidx

### ISSUE THREAD

**Issue by monsieurtanuki** [osmdroid/osmdroid#1453]
## Issue Type

- [x] Question
- [x] Improvement

## Description and/or steps/code to reproduce the problem

android.support library is kind of deprecated.
Shouldn't we roll out to [androidx](https://developer.android.com/jetpack/androidx/migrate)?
This may only impacts the demo app, not the core code.

### Version of osmdroid the issue relates to:

6.1.5

**Comment by mat8854** [comment]
Migrating is easy. There is an option in Android Studio: https://developer.android.com/jetpack/androidx/migrate

Did it for outdated applications without problems.

**Comment by InI4** [comment]
Well, I did this once and would say it was semi easy. The tool did not get everything right, manual corrections were easy, QA was heavy though.

**Comment by mat8854** [comment]
Okay, the projects where I used the migration assistant were not complex.. But the osmdroid sample application is basically just a ListView and lot of Fragments.

**Comment by monsieurtanuki** [comment]
A first step would be to switch to android support 28 in https://github.com/osmdroid/osmdroid/blob/master/gradle.properties#L44

**Comment by monsieurtanuki** [comment]
The good news is that there's no reason why the core of osmdroid (`osmdroid-android`) should be impacted by this upgrade. That was my intuition, later confirmed by a `grep "android.support"`.
The upgrade makes sense only for the activities and demos, which may use a `ToolBar` or a `FloatingActionButton`.
@mat8854 Feel like upgrading to androidx?

**Comment by mat8854** [comment]
@monsieurtanuki Yes, it is still on a different branch. An automated test suite is nice for such cases (for UI and logic). Other people have more experience with this migration task (@InI4 ).

**Comment by monsieurtanuki** [comment]
I've just tried the first step: switch android.support version to 28.0.0 (from current value: 26.1.0).
That didn't go well.
Does anybody understand the world of gradle? Help!

**Comment by InI4** [comment]
I've got no clue about gradle and I am quite offline lately, but I've managed to that occasionally.
What exactly happened after the change?

**Comment by monsieurtanuki** [comment]
I can't remember exactly, but it's immediate: when you try to sync the gradle/project after typing 28.0.0 instead of 26.1.0, there's an error message at the gradle sync level, and you can't go further.

**Comment by InI4** [comment]
I remember there were two places where you specify target and build sdk and you have to set them consistently. With recent gradle one of the settings can be removed though.

**Comment by monsieurtanuki** [comment]
Feel free to have a go with this 28.0.0 upgrade.

**Comment by InI4** [comment]
That took a bit on my end, but I now have a local build against 28.1.0 of Osmdroid 6.1.7 on my harddisk and it essentially ran fine. I still owe you a PR for something different. Hope, I find time for that.

**Comment by monsieurtanuki** [comment]
@InI4 That's good news. A PR would be perfect. Will there be little merge frictions with #1594?

**Comment by hannesa2** [comment]
I did Android X with https://github.com/osmdroid/osmdroid/pull/1599 and it's based on https://github.com/osmdroid/osmdroid/pull/1594 
But there is less progress on review side. Let's see if https://github.com/osmdroid/osmdroid/pull/1594 will be ever merged
